### ISSUE TITLE
Unable to generate OBDA from R2RML using CLI

### ISSUE THREAD

**Issue by UtkrshS** [ontop/ontop#799]
I am not able to generate the **OBDA file using the R2RML** file. It gives me an error mentioned below.

**Command used:** ./ontop mapping to-obda -i input-r2rml.ttl -o output.obda

Can someone pls help?

Error received:

it.unibz.inf.ontop.exception.InvalidOntopConfigurationException: Impossible to get the JDBC driver. Reason: No suitable driver
	at it.unibz.inf.ontop.injection.impl.OntopSQLCoreSettingsImpl.lambda$extractJdbcDriver$11(OntopSQLCoreSettingsImpl.java:121)
	at java.base/java.util.Optional.orElseGet(Optional.java:364)
	at it.unibz.inf.ontop.injection.impl.OntopSQLCoreSettingsImpl.extractJdbcDriver(OntopSQLCoreSettingsImpl.java:115)
	at it.unibz.inf.ontop.injection.impl.OntopSystemSQLSettingsImpl.loadProperties(OntopSystemSQLSettingsImpl.java:31)
	at it.unibz.inf.ontop.injection.impl.OntopSystemSQLSettingsImpl.<init>(OntopSystemSQLSettingsImpl.java:22)
	at it.unibz.inf.ontop.injection.impl.OntopStandaloneSQLSettingsImpl.loadProperties(OntopStandaloneSQLSettingsImpl.java:21)
	at it.unibz.inf.ontop.injection.impl.OntopStandaloneSQLSettingsImpl.<init>(OntopStandaloneSQLSettingsImpl.java:16)
	at it.unibz.inf.ontop.injection.impl.OntopSQLOWLAPIConfigurationImpl$BuilderImpl.build(OntopSQLOWLAPIConfigurationImpl.java:155)
	at it.unibz.inf.ontop.cli.OntopR2RMLToOBDA.run(OntopR2RMLToOBDA.java:39)
	at it.unibz.inf.ontop.cli.Ontop.main(Ontop.java:18)
Exception in thread "main" java.lang.AssertionError
	at it.unibz.inf.ontop.cli.Ontop.main(Ontop.java:30)

**Comment by LorenzBuehmann** [comment]
According to the docs, the converter

1. does consider DB metadata which it either
    a) try to extract metadata from the DB, or 
    b) expects a DB metadata file
2. or the above requirements can be ignored with `--force` argument

In case of 1a (which is the default case), Ontop has to open a JDBC connection. That means Ontop needs the corresponding JDBC driver in the classpath. The JDBC drivers of all the supported DBs aren't shipped with the Ontop distribution (due to licenses I guess),thus, you have to download the one for your DB system and put it into the `jdbc` folder of your Ontop distribution.

Long story short, either download the JDBC driver or simply bypass this with `--force`

**Comment by UtkrshS** [comment]
Thanks @LorenzBuehmann for the help, issue is resolved.

**Comment by bcogrel** [comment]
The CLI currently fails if H2 is not among the JDBC drivers (as it uses an H2 URL without specifying the JDBC driver class)

**Comment by bcogrel** [comment]
Fixed by https://gitlab.inf.unibz.it/ontop/ontop/-/commit/b04d73103e9340ca3f2e3929af39bf2d24071dcf
